### PR TITLE
renamings, consistency, comments

### PR DESCRIPTION
**Pull request by dsyme**
I suggest this approach for consistent naming

```
type AdaptiveHashSet<'T>

and aset<'T> = AdaptiveHashSet<'T>
```

and so on.  This gives scope to expand the range of adaptive structures, and also makes the meaning of `a` very obvious in documentation.

I'll propose some more renamings separately, but I think this is the core of a good adjustment

### CODE REVIEW

File: src/FSharp.Control.Incremental.Reference/ASet.fs [hunk 1]
```diff
@@ -1,150 +0,0 @@
-﻿namespace FSharp.Control.Incremental.Reference
```
Comment (dsyme) [inline src/FSharp.Control.Incremental.Reference/ASet.fs]: Note sure why Git didn't recognise the add/remove as a rename

File: src/FSharp.Control.Incremental.Tests/HashMap.fs [hunk 2]
```diff
@@ -142,14 +142,14 @@ let ``[HashMap] map2/choose2`` (lm : Map<int, int>) (rm : Map<int, int>) =
         let r = r |> Map.toList
         l = r
 
-    let add (k : int) (l : Option<int>) (r : Option<int>) =
+    let add (k : int) (l : option<int>) (r : option<int>) =
```
Comment (dsyme) [inline src/FSharp.Control.Incremental.Tests/HashMap.fs]: Generally F# code doesn't use the uppercase `Option`.

File: src/FSharp.Control.Incremental.Tests/HashSet.fs [hunk 3]
```diff
@@ -29,7 +29,7 @@ type StupidHash = { value : int } with
 
 
 /// avoid obj defaulting
-let emptyDelta : DHashSet<int> = DHashSet.empty
+let emptyDelta : HashSetDelta<int> = HashSetDelta.empty
```
Comment (dsyme) [inline src/FSharp.Control.Incremental.Tests/HashSet.fs]: Suggest dropping `D` abbreviation everywhere

File: src/FSharp.Control.Incremental/ARef/ARef.fsi [hunk 4]
```diff
@@ -1,76 +0,0 @@
-﻿namespace FSharp.Control.Incremental
```
Comment (dsyme) [inline src/FSharp.Control.Incremental/ARef/ARef.fsi]: Again the add/reove on this file wasn't recognised by git, not sure why,

**Comment by krauthaufen** [comment]
hey, cool that you keep cleaning up after me 😀
Just let me know when to merge...

**Comment by dsyme** [comment]
@krauthaufen I think it's ready to merge once latest is green

**Pull request merged**